plugins/modules/net_tools/nmcli.py

```diff
@@ -1173,6 +1451,62 @@ def show_connection(self):
 
         return conn_info
 
+    def get_supported_properties(self, setting, set_property=None, set_value='FAKEVALUE'):
+        properties = []
+
+        if setting == '802-11-wireless-security':
+            set_property = set_property or 'psk'
+
+        if set_property:
+            commands = ['set %s.%s %s' % (setting, set_property, set_value)]
+        else:
+            commands = []
```

Comment:
```suggestion
    def get_supported_properties(self, setting):
        properties = []

        if setting == '802-11-wireless-security':
            set_property = 'psk'
            set_value='FAKEVALUE'
            commands = ['set %s.%s %s' % (setting, set_property, set_value)]
        else:
            commands = []
```
I think this is over generalized at the moment. It can be refactored later if needed, but this is more clear as a special case for `'802-11-wireless-security'` IMO.

Reply:
Yes, it will very likely apply to other `properties` (if the use of this method is expanded), but the `property` currently using this exception is indeed `802-11-wireless-security`.